internal/pkg/levee/levee.go

```diff
@@ -78,6 +104,76 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+// Use the EAR pointer analysis as the propagation engine
+func runEAR(pass *analysis.Pass, state *earpointer.Partitions) (interface{}, error) {
+	conf, err := config.ReadConfig()
+	if err != nil {
+		return nil, err
+	}
+	funcSources := pass.ResultOf[source.Analyzer].(source.ResultType)
+	taggedFields := pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)
+	suppressedNodes := pass.ResultOf[suppression.Analyzer].(suppression.ResultType)
+	// Return whether a field is tainted.
+	isTaintField := func(named *types.Named, index int) bool {
+		if tt, ok := named.Underlying().(*types.Struct); ok {
+			return conf.IsSourceField(utils.DecomposeField(named, index)) ||
+				taggedFields.IsSourceField(tt, index)
+		}
+		return false
+	}
+	// Collect all the taint sources
+	for fn, sources := range funcSources {
+		// Transitively get the set of functions called within "fn".
+		// This set is used to narrow down the set of references needed to be
+		// considered during EAR heap traversal. It can also help reducing the
+		// false positives.
+		callees := make(map[*ssa.Function]bool)
+		callees[fn] = true
+		earpointer.GetCalleeFunctions(fn, callees, 7)
+
+		var fnSrcs []*source.Source
+		srcRefs := make(map[*source.Source]earpointer.ReferenceSet)
+		for _, s := range sources {
+			fnSrcs = append(fnSrcs, s)
+			srcRefs[s] = earpointer.GetSrcRefs(s, state, isTaintField, callees)
+		}
+		// Traverse all the callee functions (not just the ones with sink sources)
+		for member := range callees {
+			for _, b := range member.Blocks {
+				for _, instr := range b.Instrs {
+					switch v := instr.(type) {
+					case *ssa.Call:
+						if callee := v.Call.StaticCallee(); callee != nil &&
+							conf.IsSink(utils.DecomposeFunction(callee)) {
```

Comment:
I think this would read better on a single line. In general, it is best to avoid splitting conditionals.

Same comment for below lines.